gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/computer/traversal/step/map/PageRankVertexProgramStep.java

```diff
@@ -83,17 +83,17 @@ public String toString() {
     }
 
     @Override
-    public PageRankVertexProgram generateProgram(final Graph graph, final Memory memory) {
+    public PageRankVertexProgram generateProgram(final Memory memory, final Graph... graphs) {
         final Traversal.Admin<Vertex, Edge> detachedTraversal = this.edgeTraversal.getPure();
-        detachedTraversal.setStrategies(TraversalStrategies.GlobalCache.getStrategies(graph.getClass()));
+        detachedTraversal.setStrategies(TraversalStrategies.GlobalCache.getStrategies(graphs[0].getClass()));
```

Comment:
I've not thought this through, but I think we would only ever have one `Graph` instance in this case - should that be validated at all?

Reply:
Since it's called through a step, I think you're right that there should only be one `Graph`. I will validate in the "steps" that use the `Graph`.